test-storybooks/portable-stories-kitchen-sink/nextjs/stories/NextHeader.stories.tsx

```diff
@@ -0,0 +1,28 @@
+import type { Meta } from '@storybook/react';
+import type { StoryObj } from '@storybook/react';
+import { expect, userEvent, within } from '@storybook/test';
+import { cookies, headers } from '@storybook/nextjs/headers';
+import NextHeader from './NextHeader';
+
+export default {
+  component: NextHeader,
+} as Meta<typeof NextHeader>;
+
+type Story = StoryObj<typeof NextHeader>;
+
+export const Default: Story = {
+  loaders: async () => {
+    cookies().set('firstName', 'Jane');
+    cookies().set({
+      name: 'lastName',
+      value: 'Doe',
+    });
+    headers().set('timezone', 'Central European Summer Time');
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    const submitButton = await canvas.findByRole('button');
+    await userEvent.click(submitButton);
+    expect(headers().get('cookie')).toContain('user-id=encrypted-id');
```

Comment:
1. Should we also assert all the cookies+headers get+set+getAll calls to have been made?
2. How about asserting on the UI actually containing the data we set in the loaders? I think @kasperpeulen's point is that is automatically asserted with VRT, but then at least we could have a description above the story explaining it I think.

Reply:
Good one! I made the changes. This story is in two places, one for portable stories and another for the sandbox. The sandbox one will VRT